report_custom_filename/controllers/reports.py

```diff
@@ -16,10 +16,9 @@ def index(self, action, token):
         action = json.loads(action)
         context = dict(http.request.context)
         context.update(action["context"])
-        report_xml = http.request.session.model('ir.actions.report.xml')
+        report_xml = http.request.env['ir.actions.report.xml']
         reports = report_xml.search(
-            [('report_name', '=', action['report_name'])],
-            0, False, False, context)
+            [('report_name', '=', action['report_name'])])
```

Comment:
As you're touching this, you can include here the condition `('download_filename', '!=', False)` to avoid the later `if`